io/buf_reader.ts

```diff
@@ -26,10 +26,12 @@ export class BufferFullError extends Error {
  * @deprecated This will be removed in 1.0.0. Use the {@link https://developer.mozilla.org/en-US/docs/Web/API/Streams_API | Web Streams API} instead.
  */
 export class PartialReadError extends Error {
-  override name = "PartialReadError";
-  partial?: Uint8Array;
-  constructor() {
+  partial: Uint8Array;
+
+  constructor(partial: Uint8Array) {
```

Comment:
Previously, this property would be set within `try`/`catch` blocks. It should instead just be set during construction (see below).